chapters/solutions.Rmd

```diff
@@ -1165,11 +1163,76 @@ parser.add_argument('--style', type=str, nargs='*', choices=plt.style.available,
 
 ### Exercise \@ref(config-ex-saveload) {-}
 
-FIXME: GVW to write solution
+The first step is to add a new command-line argument to tell `plotcount.py` what we want to do:
+
+```python
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description=__doc__)
+    # ...other options as before...
+    parser.add_argument('--saveconfig', type=str, default=None,
+                        help='Save configuration to file')
+    args = parser.parse_args()
+    main(args)
+```
+
+Next, we add three lines to `main` to act on this option *after* all of the plotting parameters have been set.
+For now we use `return` to exit from `main` as soon as the parameters have been saved;
+this lets us test our change without overwriting any of our actual plots.
+
+```python
```

Comment:
In this code block where you define the updated `main` function you also need to define the new `save_configuration` function.

```
def save_configuration(filename, params):
    """Save configuration to a file."""
    with open(filename, 'w') as reader:
        yaml.dump(params, reader)
```